.github/workflows/auto-dependabot.yml

```diff
@@ -0,0 +1,31 @@
+name: auto-dependabot-fix
+
+on:
+  pull_request_target: ~
+
+jobs:
+  building:
+    if: github.actor == 'dependabot[bot]' && !github.event.pull_request.head.repo.fork
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v5
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+          token: ${{ secrets.GH_CLOUD_SDK_JS_ADMIN_WRITE_TOKEN }}
+      - run: git fetch --depth=1
+      - uses: actions/setup-node@v5
+        with:
+          node-version: 22
+          cache: 'yarn'
+      - name: Install Dependencies & Compile
+        run: yarn install --frozen-lockfile --ignore-engines
```

Comment:
[req] Use `pnpm` instead of `yarn`. Consider using the reusable `setup` action instead.